### ISSUE TITLE
Provide version 4.7 of nuget.commandline package

### ISSUE THREAD

**Issue by pascalberger** [NuGet/Home#7114]
4.7 became the recommended latest version, but [nuget.commandline package](https://www.nuget.org/packages/NuGet.CommandLine/) is still at 4.6.2.

Can you please update the package to 4.7.

Would it also be possible to have this package updated as part of your regular build chain to have it always updated to the latest recommended version?

**Comment by PatoBeltran** [comment]
I know you are on this @nkolev92

**Comment by nkolev92** [comment]
I just pushed it. 
It should be available shortly. 

/cc @rrelyea. 

@pascalberger 
Updating it as part of the regular build chain is not a good idea as that's not stable. 

The contract for update self is that will update to the latest stable. 

We won't allow it to update to prerelease before first enhancing the command. 

Related: 
https://github.com/NuGet/Home/issues/4197
https://github.com/NuGet/Home/issues/1783

Since our latest build is available at https://dotnet.myget.org/F/nuget-build/api/v2/package/NuGet.CommandLine, once the latter issue is finished you could just do 
`update -self source https://dotnet.myget.org/F/nuget-build/api/v2/`

**Comment by pascalberger** [comment]
@nkolev92 Thanks for the update.

To give some context, I'm deploying NuGet through Chocolatey. The [Chocolatey Package](https://chocolatey.org/packages/NuGet.CommandLine) is updated whenever the NuGet package is updated which is why I need this NuGet package to be up to date.

Also is there a reason to have it now using a SemVer2 version number containing metadata which will break access to it from any non v3 compatible clients?

**Comment by nkolev92** [comment]
@pascalberger 
Do you really need unstable builds in that case? 
I think depending on whatever is published on nuget.org is the correct approach here as that's a stable and vetted release vs depending on the nightly build which might have unforeseen regressions. 

Regarding the semver2 metadata, I created an issue about that:
https://github.com/NuGet/Home/issues/7116

We do that with the rest of our packages https://www.nuget.org/packages/NuGet.Frameworks/4.8.0-preview3.5278, and we don't think it's unreasonable to expect customers to use the newer clients. 

Personally not 100% confident that's the right approach for the nuget.commandline nupkg as that's using during nuget update -self

**Comment by pascalberger** [comment]
@nkolev92 I didn't ask for unstable nightly builds to be published, but for a reliable way to have the nuget.commandline package updated to latest version, as I had to request an update to latest recommended version for several times in the past (https://github.com/NuGet/Home/issues/5784, https://github.com/NuGet/Home/issues/6203). Therefore it would be nice to have the package automatically updated if a new recommended latest version was released.

**Comment by nkolev92** [comment]
@pascalberger 

For 
> Therefore it would be nice to have the package automatically updated if a new recommended latest version was released.

please refer to the following issue: https://github.com/NuGet/Home/issues/4851

//cc @rrelyea

**Comment by pascalberger** [comment]
@nkolev92 #4851 makes perfect sense. My point was that the points below point 5 were not always made at the same time in the past, also for example for 4.7 leading to this issue, and therefore my suggestion to automate this step to not forget it to publish to nuget.org.
